compiler/util/files.cpp

```diff
@@ -638,110 +638,193 @@ void genIncludeCommandLineHeaders(FILE* outfile) {
   }
 }
 
+std::string genMakefileEnvCache(void);
+std::string genMakefileEnvCache(void) {
+  std::string result;
+  std::map<std::string, const char*>::iterator env;
 
-void codegen_makefile(fileinfo* mainfile, const char** tmpbinname, bool skip_compile_link, const std::vector<const char*>& splitFiles) {
-  fileinfo makefile;
-  openCFile(&makefile, "Makefile");
+  for (env = envMap.begin(); env != envMap.end(); ++env) {
+    const std::string& key = env->first;
+    const char* oldPrefix = "CHPL_";
+    const char* newPrefix = "CHPL_MAKE_";
+    INT_ASSERT(key.substr(0, strlen(oldPrefix)) == oldPrefix);
+    std::string keySuffix = key.substr(strlen(oldPrefix), std::string::npos);
+    std::string chpl_make_key = newPrefix + keySuffix;
+    result += chpl_make_key + "=" + std::string(env->second) + "|";
+  }
+
+  return result;
+}
+
+void codegen_makefile(fileinfo* mainfile, const char** tmpbinname,
```

Comment:
Oof.  We had another miscommunication here, and I believe this one cost you a lot of time.  My intention was for the shared code of `codegen_makefile` and `codegen_makefile_mli` to be moved into helper functions that could be called from both rather than making one gargantuan function that handled both strategies

Reply:
I rather like how it turned out and I don't think it actually took too much time!

Reply:
I think that the two functions share enough common code that breaking the new `codegen_makefile()` up into a series of helper functions would actually end up adding more (only called once) code than the current approach.

Reply:
Okay, that's fair